graphene_django/static/graphene_django/graphiql.js

```diff
@@ -61,17 +61,29 @@
   var fetchURL = locationQuery(otherParams);
 
   // Defines a GraphQL fetcher using the fetch API.
-  function httpClient(graphQLParams) {
-    var headers = {
-      Accept: "application/json",
-      "Content-Type": "application/json",
-    };
+  function httpClient(graphQLParams, opts = { headers: {} }) {
+    let headers = opts.headers;
+    // Convert headers to an object.
+    if (typeof headers === 'string') {
+      headers = JSON.parse(opts.headers);
+    }
     if (csrftoken) {
-      headers["X-CSRFToken"] = csrftoken;
+      Object.assign(
+          {
+            'X-CSRFToken': csrftoken
+          },
+          headers,
+      )
```

Comment:
`Object.assign` does 2 things:
1. mutates the `target` (first param) object with `source` (second param) by doing a shallow comparison
2. returns the mutated object (target)

So in this case, you'd mutate an object literal and then return it, but the returned value is not used in any way, so the whole operation is essentially useless. I think this change is rather unnecessary in what you want to achieve.

Also, `Object.assign` is available from es6+, so you can't use that here probably anyway.